identity-account-storage/src/storage/traits.rs

```diff
@@ -27,45 +29,88 @@ mod storage_sub_trait {
+  ///
+  /// This operation is idempotent: it does not fail if the given `did` does not (or no longer) exist.
+  ///
+  /// Returns `true` if the did and its associated data was removed, `false` if nothing was done.
+  async fn did_purge(&self, did: &IotaDID) -> Result<bool>;
+
+  /// Generates a new key for the given `did` with the given `key_type` and `fragment` identifier
+  /// and returns the location of the newly generated key.
+  async fn key_generate(&self, did: &IotaDID, key_type: KeyType, fragment: &str) -> Result<KeyLocation>;
+
+  /// Inserts a private key at the specified `location`.
+  ///
+  /// If a key at `location` exists, it is overwritten.
+  async fn key_insert(&self, did: &IotaDID, location: &KeyLocation, private_key: PrivateKey) -> Result<()>;
+
+  /// Retrieves the public key from `location`.
+  async fn key_public(&self, did: &IotaDID, location: &KeyLocation) -> Result<PublicKey>;
+
+  /// Deletes the key at `location`.
+  ///
+  /// This operation is idempotent: it does not fail if the key does not exist.
+  ///
+  /// Returns `true` if it removed the key, `false` if nothing was done.
+  async fn key_delete(&self, did: &IotaDID, location: &KeyLocation) -> Result<bool>;
 
   /// Signs `data` with the private key at the specified `location`.
   async fn key_sign(&self, did: &IotaDID, location: &KeyLocation, data: Vec<u8>) -> Result<Signature>;
 
-  /// Returns `true` if a keypair exists at the specified `location`.
+  /// Returns `true` if a key exists at the specified `location`.
   async fn key_exists(&self, did: &IotaDID, location: &KeyLocation) -> Result<bool>;
 
   /// Returns the chain state of the identity specified by `did`.
-  async fn chain_state(&self, did: &IotaDID) -> Result<Option<ChainState>>;
+  async fn chain_state_get(&self, did: &IotaDID) -> Result<Option<ChainState>>;
 
   /// Set the chain state of the identity specified by `did`.
-  async fn set_chain_state(&self, did: &IotaDID, chain_state: &ChainState) -> Result<()>;
+  async fn chain_state_set(&self, did: &IotaDID, chain_state: &ChainState) -> Result<()>;
 
-  /// Returns the state of the identity specified by `did`.
-  async fn state(&self, did: &IotaDID) -> Result<Option<IdentityState>>;
+  /// Returns the [`IotaDocument`] of the identity specified by `did`.
+  async fn document_get(&self, did: &IotaDID) -> Result<Option<IotaDocument>>;
 
   /// Sets a new state for the identity specified by `did`.
-  async fn set_state(&self, did: &IotaDID, state: &IdentityState) -> Result<()>;
+  async fn document_set(&self, did: &IotaDID, state: &IotaDocument) -> Result<()>;
+
+  /// Returns `true` if the index contains the given `did`.
+  async fn index_has(&self, did: &IotaDID) -> Result<bool>;
 
-  /// Removes the keys and any state for the identity specified by `did`.
-  async fn purge(&self, did: &IotaDID) -> Result<()>;
+  /// Retrieves the list of [`IotaDID`]s stored in the index.
+  async fn index(&self) -> Result<Vec<IotaDID>>;
```

Comment:
Maybe call this `did_index` or `did_list`? Otherwise it stands as an outlier with no other common functions if `index_has` is renamed.

Reply:
Great suggestion, thanks. This makes the interface so much more consistent :rocket: